readthedocs/search/management/commands/reindex_elasticsearch.py

```diff
@@ -0,0 +1,95 @@
+import datetime
+import logging
+
+from celery import chord, chain
+from django.apps import apps
+from django.conf import settings
+from django.core.management import BaseCommand
+from django_elasticsearch_dsl.registries import registry
+
+from ...tasks import (index_objects_to_es_task, switch_es_index_task, create_new_es_index_task,
+                      index_missing_objects_task)
+from ...utils import chunks
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+
+    @staticmethod
+    def _get_indexing_tasks(app_label, model_name, instance_ids, document_class, index_name):
+        chunk_objects = chunks(instance_ids, settings.ES_TASK_CHUNK_SIZE)
+
+        for chunk in chunk_objects:
+            data = {
+                'app_label': app_label,
+                'model_name': model_name,
+                'document_class': document_class,
+                'index_name': index_name,
+                'objects_id': chunk
+            }
+            yield index_objects_to_es_task.si(**data)
+
+    def _run_reindex_tasks(self, models):
+        for doc in registry.get_documents(models):
+            qs = doc().get_queryset()
+            instance_ids = list(qs.values_list('id', flat=True))
```

Comment:
Is this not going to use a ton of memory? This is creating the entire list of objects in memory all at once, instead of streaming them.

Reply:
Its not actually creating list of objects. Its just a list of integers. I think integers take much lower amount of memory.

Reply:
I'm pretty worried this is going to be both slow & perhaps incredibly memory intensive in production when we have 2 million objects in memory here. I'd like to find another approach that streams the data if possible, but we can try this for now and see what happens.

Reply:
@ericholscher I think we can try this now to check if anything wrong happens.

Reply:
@ericholscher I have optimized the memory usage in 143ce7f